src/commands/auth/list.ts

```diff
@@ -25,24 +26,35 @@ const MAYBE = '   -?-  '
 export default class AuthList extends NimBaseCommand {
   static description = 'List all your Nimbella namespaces'
 
-  static flags: typeof NimBaseCommand.flags = {
+  static flags = {
+    apihost: flags.string({ description: 'Only list namespaces for the specififed API host' }),
     ...NimBaseCommand.flags
   }
 
   static args = []
 
   async runCommand(rawArgv: string[], argv: string[], args: any, flags: any, logger: NimLogger): Promise<void> {
     const dict = await getCredentialDict(authPersister)
-    if (Object.keys(dict).length === 1) {
-      await this.formatCredentialList(Object.values(dict)[0], logger)
-    } else {
-      for (const host in dict) {
-        const rows = dict[host]
-        if (rows.length === 0) continue
-        logger.log('')
+    if (flags.apihost) {
+      const host = parseAPIHost(flags.apihost)
+      if (host in dict) {
         logger.log(`${bold(`On API host ${host}:`)}`)
         await this.formatCredentialList(dict[host], logger)
       }
+    } else {
+      if (Object.keys(dict).length === 1) {
+        await this.formatCredentialList(Object.values(dict)[0], logger)
```

Comment:
I wasn't sure why this is here as a special case - but I admit to not trying to understand. I didn't really modify the existing logic when `--apihost` is omitted. There is a small change in formatting though: the leading line that's printed when `nim auth list` is run is not removed.